src/Microsoft.SqlTools.ManagedBatchParser/BatchParser/ExecutionEngineCode/ScriptExecutionArgs.cs

```diff
@@ -130,11 +130,7 @@ internal int StartingLine
         {
             get
             {
-                if (cmdVariables == null)
-                {
-                    cmdVariables = new Dictionary<string, string>(StringComparer.CurrentCultureIgnoreCase);
-                }
-
+                cmdVariables ??= new Dictionary<string, string>(StringComparer.CurrentCultureIgnoreCase);
                 return cmdVariables;
```

Comment:
Maybe I'm missing something, but... do we really think we save much with "a lazy init" of the backing field?

I would just:
1) Drop the backing field at line 18
2) Declare the internal property like this:

```csharp
   internal Dictionary<string, string> Variables { get;  } = new Dictionary<string, string>(StringComparer.CurrentCultureIgnoreCase);
```

(I would have make it return an IDictionary, but then again... we love "internal" accessors in our codebases, so we could break somebody...)

Reply:
We have a lot of opportunities here, I agree. I'll consider that for next time :)